docs: decouple installation & add stability section (#811)
hashicorp/terraform-ls: README.md

# Terraform Language Server

> `terraform-ls` is the official [Terraform](https://www.terraform.io) language server maintained by [HashiCorp](https://www.hashicorp.com). It provides IDE features to any [LSP](https://microsoft.github.io/language-server-protocol/)-compatible editor.

Reviewer: ```suggestion
The official [Terraform](https://www.terraform.io) language server (`terraform-ls`) maintained by [HashiCorp](https://www.hashicorp.com) provides IDE features to any [LSP](https://microsoft.github.io/language-server-protocol/)-compatible editor.
```

and/or [open new issue](https://github.com/hashicorp/terraform-ls/issues/new/choose)
if you experience a bug or have an idea for a feature.

## Stability

We aim to communicate our intentions regarding breaking changes via [semver](https://semver.org). Relatedly we may use pre-releases, such as `MAJOR.MINOR.PATCH-beta1` to gather early feedback on certain features and changes.

We ask that you [report any bugs](https://github.com/hashicorp/terraform-ls/issues/new/choose) in any versions but especially in pre-releases, if you decide to use them.

## Installation

> Some editors have built-in logic to install and update the language server automatically, so you may not need to worry about installation or updating of the server.

Reviewer: ```suggestion
Most text editor extensions/plugins have built-in logic to install and update language servers automatically, so you typically shouldn't need to worry about installation or updating of the server, as long as you use the provided extension.
```

> Read the [installation page](./docs/installation.md) for more.

Reviewer: ```suggestion
Read the [installation page](./docs/installation.md) for installation instructions.
```
